libs/langchain/langchain/schema/storage.py

```diff
@@ -0,0 +1,53 @@
+from abc import ABC, abstractmethod
+from typing import Generic, Iterator, List, Optional, Sequence, Tuple, TypeVar, Union
+
+K = TypeVar("K")
+V = TypeVar("V")
+
+
+class BaseStore(Generic[K, V], ABC):
```

Comment:
worth implementing MutableMapping for convenience? seems like you could cover all the builtin methods with the abstract methods already included

Reply:
This we can add later if we think it's okay. I don't want to make it look too much like a dictionary though i'd rather folks used it through mset and mget, so the method names are explicit and so that batch is always favored; e.g., I don't think we should expose an interface to make it convenient to get or write a key at a time